### ISSUE TITLE
Sync process fails when Sync QR code is scanned if app has been in backgroun

### ISSUE THREAD

**Issue by VolodLytvynenko** [status-im/status-react#19452]
### Steps to Reproduce:
1. `Device A` is in the background for at least 5 minutes.
2. `Device A` generates a sync QR code.
3. `Device B` scans the QR code during onboarding.

### Actual Result:
The sync process fails for Device B, showing a 'failed to connect to any of given addresses' error message in a toast.

![image](https://github.com/status-im/status-mobile/assets/52490791/67ad5f90-686a-4f39-945f-9f24003582c3)

### Notes:
-  Generating a new QR, relogin for  `Device A` and reopening the app did not resolve the issue.
- Only after apps were reinstalled on  `Device A` and  `Device B` resolved the issue.

### Logs:
[logs.txt](https://github.com/status-im/status-mobile/files/14805111/logs.txt)

**Comment by cammellos** [comment]
> Generating a new QR, relogin for Device A and reopening the app did not resolve the issue.

This should be addressed, the rest is low priority (it's ok that a QR code doesn't work if the app goes to background, but regenerating should work)

**Comment by cammellos** [comment]
@pavloburykh mentioned that he wasn't able to replicate, if you can't please pnig @VolodLytvynenko

**Comment by VolodLytvynenko** [comment]
> @pavloburykh mentioned that he wasn't able to replicate, if you can't please pnig @VolodLytvynenko

@cammellos can't replicate it anymore. Probably my device switched to a mobile network somehow even when wifi was enabled.. Can be closed
